packages/auth-construct/src/construct.ts

```diff
@@ -700,7 +700,7 @@ export class AmplifyAuth
       );
       result.oAuthMappings[authProvidersList.amazon] =
         external.loginWithAmazon.clientId;
-      result.providersList.push('AMAZON');
+      result.providersList.push('LOGIN_WITH_AMAZON');
```

Comment:
is there some way we can get the type system to fail incorrect values at build time here?

Reply:
The best way to catch this in the future will be in some form of integration test, by reading amplify_config.json and comparing it to the config schema.

https://github.com/aws-amplify/amplify-backend/blob/main/packages/client-config/src/client-config-schema/schema_v1.json

Reply:
Perhaps we should go ahead and add that test? We could use [ajv](https://www.npmjs.com/package/ajv) to validate that the client config matches the schema

Reply:
I can follow up and try adding a test for the JSON validation.